### PR TITLE
Single coord

### PR DESCRIPTION
**Pull request by debebantur**
This will help when coordinator is on different host.  If coordinator moves to router host (https://github.com/pg-sharding/spqr/issues/279) this pr will still help

### CODE REVIEW

**Comment by reshke** [comment]
>If coordinator moves to router host (https://github.com/pg-sharding/spqr/issues/279) this pr can be closed

This is not true. 
If coordinator run logic moves to separate router goroutine instead of own process, nothing changes.

File: qdb/etcdqdb.go [hunk 1]
```diff
@@ -52,6 +53,10 @@ const (
 	routersNamespace       = "/routers"
 	shardingRulesNamespace = "/sharding_rules"
 	shardsNamespace        = "/shards"
+	//must be more or equal to 5
```
Comment (Denchick) [inline qdb/etcdqdb.go:56]: Why?

File: coordinator/provider/coordinator.go [hunk 2]
```diff
@@ -256,22 +256,39 @@ func (qc *qdbCoordinator) watchRouters(ctx context.Context) {
...
-// spqr router`s availability
 func NewCoordinator(db qdb.XQDB) *qdbCoordinator {
-	cc := &qdbCoordinator{
+	return &qdbCoordinator{
 		db: db,
 	}
+}
+
+// RunCoordinator side efferc: runs async goroutine that checks
+// spqr router`s availability
```
Comment (Denchick) [inline coordinator/provider/coordinator.go:266]: // RunCoordinator side effect: it runs an asynchronous goroutine 
// that checks the availability of the SPQR router

File: coordinator/provider/coordinator.go [hunk 3]
```diff
@@ -256,22 +256,39 @@ func (qc *qdbCoordinator) watchRouters(ctx context.Context) {
...
+func (cc *qdbCoordinator) RunCoordinator(ctx context.Context) {
+	if cc.db.TryCoordinatorLock(context.TODO()) != nil {
+	LOCK:
```
Comment (Denchick) [inline coordinator/provider/coordinator.go:269]: the wat
Lets rewrite this method without using LOCK tag
Or just move TryCoordinatorLock to a separate method

**Review by Denchick — CHANGES_REQUESTED** [review]

**Comment by reshke** [comment]
LGTM?

**Review by Denchick — APPROVED** [review]

**Pull request merged**